### PR TITLE
Further optimize the download of SPDX license texts

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/spdx/build.gradle.kts [hunk 1]
```diff
@@ -286,30 +286,16 @@ fun generateEnumClass(
...
-                if (!downloadedFile.renameTo(targetFile)) {
-                    throw GradleException("Failed to rename file '$downloadedFile' to '$targetFile'.")
-                }
+        val resourcePaths = listOf("licenses", "exceptions").map { file("src/main/resources/$it") }
+
+        resourcePaths.forEach { path ->
+            path.listFiles().forEach { file ->
+                // Trim trailing whitespace and blank lines.
+                val lines = file.readLines().map { it.trimEnd() }.asReversed().dropWhile { it.isEmpty() }
+                    .asReversed().dropWhile { it.isEmpty() }
```
Comment (mnonnenmacher) [inline utils/spdx/build.gradle.kts:296]: Can't you use `dropLastWhile` instead of reversing?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: utils/spdx/build.gradle.kts [hunk 2]
```diff
@@ -286,30 +286,16 @@ fun generateEnumClass(
...
-
-                if (!downloadedFile.renameTo(targetFile)) {
-                    throw GradleException("Failed to rename file '$downloadedFile' to '$targetFile'.")
-                }
+        val resourcePaths = listOf("licenses", "exceptions").map { file("src/main/resources/$it") }
+
+        resourcePaths.forEach { path ->
+            path.listFiles().forEach { file ->
+                // Trim trailing whitespace and blank lines.
+                val lines = file.readLines().map { it.trimEnd() }.asReversed().dropWhile { it.isEmpty() }
```
Comment (fviernau) [inline utils/spdx/build.gradle.kts:295]: I needed to look a while at lines 294 - 296 to understand what it's doing, and I still don't understand why `asReversed` is used (twice). I believe factoring out that code into a function with a speaking name would solve that problem and make the code comment (at least in this place) redundant.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: utils/spdx/build.gradle.kts (hunk 2 above)
Comment (sschuberth) [reply utils/spdx/build.gradle.kts:295]: Please note that this code has been there for quite some time and was just moved multiple times recently. I'll still have a look how to improve it, but in a new commit.

Comment (fviernau) [reply utils/spdx/build.gradle.kts:295]: Oh sorry, I missed that. Please disregard my comment then.

**Review by fviernau — APPROVED** [review]

**Review by fviernau — APPROVED** [review]

File: utils/spdx/build.gradle.kts [hunk 3]
```diff
@@ -286,30 +286,16 @@ fun generateEnumClass(
...
-            // Rename files that are named differently on the server.
-            if (downloadedFile != targetFile) {
-                if (targetFile.exists() && !targetFile.delete()) {
-                    throw GradleException("Failed to delete existing file '$targetFile'.")
-                }
-
-                if (!downloadedFile.renameTo(targetFile)) {
-                    throw GradleException("Failed to rename file '$downloadedFile' to '$targetFile'.")
-                }
+        val resourcePaths = listOf("licenses", "exceptions").map { file("src/main/resources/$it") }
```
Comment (mnonnenmacher) [inline utils/spdx/build.gradle.kts:290]: Commit message nit: You don't mention which plugin you mean.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: utils/spdx/build.gradle.kts (hunk 3 above)
Comment (mnonnenmacher) [reply utils/spdx/build.gradle.kts:290]: Sorry, now you have a typo in the plugin name.

Comment (sschuberth) [reply utils/spdx/build.gradle.kts:290]: Fixed, I was doing to many things at a time it seems.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**